Lib/logging/__init__.py

```diff
@@ -1880,50 +1881,56 @@ def process(self, msg, kwargs):
     #
     # Boilerplate convenience methods
     #
-    def debug(self, msg, *args, **kwargs):
+    def debug(self, msg, *args, stacklevel=1, **kwargs):
         """
         Delegate a debug call to the underlying logger.
         """
-        self.log(DEBUG, msg, *args, **kwargs)
+        self._log(DEBUG, msg, *args, stacklevel=stacklevel + 1, **kwargs)
 
-    def info(self, msg, *args, **kwargs):
+    def info(self, msg, *args, stacklevel=1, **kwargs):
         """
         Delegate an info call to the underlying logger.
         """
-        self.log(INFO, msg, *args, **kwargs)
+        self._log(INFO, msg, *args, stacklevel=stacklevel + 1, **kwargs)
 
-    def warning(self, msg, *args, **kwargs):
+    def warning(self, msg, *args, stacklevel=1, **kwargs):
         """
         Delegate a warning call to the underlying logger.
         """
-        self.log(WARNING, msg, *args, **kwargs)
+        self._log(WARNING, msg, *args, stacklevel=stacklevel + 1, **kwargs)
 
-    def error(self, msg, *args, **kwargs):
+    def error(self, msg, *args, stacklevel=1, **kwargs):
         """
         Delegate an error call to the underlying logger.
         """
-        self.log(ERROR, msg, *args, **kwargs)
+        self._log(ERROR, msg, *args, stacklevel=stacklevel + 1, **kwargs)
 
-    def exception(self, msg, *args, exc_info=True, **kwargs):
+    def exception(self, msg, *args, stacklevel=1, exc_info=True, **kwargs):
         """
         Delegate an exception call to the underlying logger.
         """
-        self.log(ERROR, msg, *args, exc_info=exc_info, **kwargs)
+        self._log(ERROR, msg, *args, stacklevel=stacklevel + 1, exc_info=exc_info, **kwargs)
 
-    def critical(self, msg, *args, **kwargs):
+    def critical(self, msg, *args, stacklevel=1, **kwargs):
         """
         Delegate a critical call to the underlying logger.
         """
-        self.log(CRITICAL, msg, *args, **kwargs)
+        self._log(CRITICAL, msg, *args, stacklevel=stacklevel + 1, **kwargs)
 
-    def log(self, level, msg, *args, **kwargs):
+    def log(self, level, msg, *args, stacklevel=1, **kwargs):
         """
         Delegate a log call to the underlying logger, after adding
         contextual information from this adapter instance.
         """
+        self._log(level, msg, *args, stacklevel=stacklevel + 1, **kwargs)
+
+    def _log(self, level, msg, *args, stacklevel=1, **kwargs):
```

Comment:
It will be more efficient to pass `args`, without packing/unpacking.
```suggestion
    def _log(self, level, msg, args, stacklevel=1, **kwargs):
```